HIRS_Utils/src/main/java/hirs/data/persist/ReferenceManifest.java

```diff
@@ -223,7 +228,11 @@ public ReferenceManifest(final byte[] rimBytes) throws IOException {
 
         this.rimBytes = rimBytes.clone();
 
-        SoftwareIdentity si = validateSwidTag(new ByteArrayInputStream(rimBytes));
+        SwidTagValidator validator = new SwidTagValidator();
+        validator.validateSwidtagInputStream(new ByteArrayInputStream(rimBytes));
+        SoftwareIdentity si = validator.getSoftwareIdentity();
+        payloadFileValid = validator.isPayloadFileValid();
```

Comment:
If this is to compare the file hash in the xml swidtag, please look at the code at the bottom of the constructor.  This is already stored in rimHash.  This is mainly a note for later but going down into SwidTagValidator to compare this is unnecessary and non-functional with the changes I'm making.  The main problem is that this constructor is only for base RIMs and the validation won't work if the support rim isn't uploaded first, which isn't a guarantee.